upgrade/steps.go

```diff
@@ -494,6 +494,10 @@ func EnsureBranchCheckedOut(ctx Context, branchName string) step.Step {
 //
 // The second argument represents a message to describe the result. It may be empty.
 func GetExpectedTarget(ctx Context, name string) ([]UpgradeTargetIssue, string, error) {
+	if ctx.TargetVersion != nil {
+		return []UpgradeTargetIssue{{Version: ctx.TargetVersion, Number: 0}}, "", nil
```

Comment:
This function returns a list of issues that the upgrade will close: `[]UpgradeTargetIssue`. Since the list of issues we are closing is no longer directly tied to the upstream upgrade target like it used to be, can we change the function signature so it accurately represents the data we are returning.

Maybe something like:
```go
type UpstreamUpgradeTarget struct {
    // The version we are targeting. `nil` indicates that no upstream upgrade was found.
    Version *semver.Version
    // The list of issues that this upgrade will close.
    GHIssues []UpgradeTargetIssue
}
```